migrations/2021-07-20-102033_actor_name_length/down.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE community ALTER COLUMN name TYPE varchar(20);
+ALTER TABLE person ALTER COLUMN name TYPE varchar(20);
```

Comment:
This fails with `cannot alter type of a column used by a view or rule`. From what I found, in this case it is necessary to drop the views and recreate them after. That seems quite complicated as we have a lot of views. Is there any easier way?

Reply:
That is unavoidable, but I got rid of nearly every view when I moved them to diesel so this shouldn't be difficult. 

There are only 3 views now:

```
lemmy=# \dv
            List of relations
 Schema |      Name       | Type | Owner
--------+-----------------+------+-------
 public | comment_alias_1 | view | lemmy
 public | person_alias_1  | view | lemmy
 public | person_alias_2  | view | lemmy
```

grep the migrations folder (sort by recent) for that view name and you'll see how to regenerate them. They're necessary to get around a diesel limitation of not being able to do self joins.

Reply:
Okay I didnt even check the view definitions before, assumed they would be much more complicated than that.